cmd/run.go

```diff
@@ -263,13 +264,16 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) error {
 			logger.WithError(klErr).Warn("Error while closing the SSLKEYLOGFILE")
 		}
 	}
-	if err != nil {
-		return errext.WithExitCodeIfNone(common.UnwrapGojaInterruptedError(err), exitcodes.GenericEngine)
-	}
+
 	if engine.IsTainted() {
-		return errext.WithExitCodeIfNone(errors.New("some thresholds have failed"), exitcodes.ThresholdsHaveFailed)
+		if err == nil {
+			err = errors.New("some thresholds have failed")
+		} else {
+			logger.Error("some thresholds have failed") // log this, even if there was already a previous error
```

Comment:
Just out of the curiosity, why does it look like this and not like:

```golang
err = fmt.Errorf("%w; some thresholds have failed", err)
```

Reply:
No strong reason, just 2 weak ones :sweat_smile: First, because the two errors are not actually connected, I don't think it makes sense to wrap one in the other and make a composite error :thinking: 

Second reason is that it is probably going to be a bit more visible to the user if `some thresholds have failed` is a separate `error` log message to the error message that is stored in `err` and will actually abort the test. It could be quite long, so `some thresholds have failed` might get lost at its end

Reply:
I'll buy it